registry/server/api_topics.go

```diff
@@ -185,25 +185,37 @@ func (s *Server) CreateTopic(ctx context.Context, req *pb.CreateTopicRequest) (*
 	// If we're targeting a specific set of brokers by tag, build
 	// a replica assignment.
 	var assignment kafkaadmin.ReplicaAssignment
-	if req.TargetBrokerTags != nil {
+	if req.TargetBrokerTags != nil || req.TargetBrokerIds != nil {
 		// Create a stub map with the provided request dimensions.
 		opts := kafkazk.Populate(
 			req.Topic.Name,
 			int(req.Topic.Partitions),
 			int(req.Topic.Replication),
 		)
 		pMap := kafkazk.NewPartitionMap(opts)
-
 		// Fetch brokers by tag.
 		reqParams := &pb.BrokerRequest{Tag: req.TargetBrokerTags}
 		resp, err := s.ListBrokers(ctx, reqParams)
 		if err != nil {
 			return empty, err
 		}
 
-		targetBrokerIDs := make([]int, len(resp.Ids))
-		for i := range resp.Ids {
-			targetBrokerIDs[i] = int(resp.Ids[i])
+		selectedBrokerIds := make(map[uint32]bool)
+		if req.TargetBrokerIds != nil {
+			for _, id := range req.TargetBrokerIds {
+				selectedBrokerIds[id] = true
+			}
+		}
+
+		var targetBrokerIDs []int
+		for _, id := range resp.Ids {
+			if len(selectedBrokerIds) > 0 {
```

Comment:
Since the map was initialized with `selectedBrokerIds := make(map[uint32]bool)`, the map itself is non-nil, so similarly to the above, we won't seen an error here (the `_, ok :=` pattern on non-nil maps but nil keys is safe).

Reply:
Ignore my last comment; at the first read, I didn't catch that we _want_ a list intersection here. I think the code should have some comments describing the conditionals (ie that we're accepting both brokers by tags and/or brokers by ID, if we specify both, that the results are intersected).